Adding a simple VoC increase to Reff & setting a fixed end date to second wave

tobinsouth opened the pull request:
Early commits add at Beta(6,14)+1 increase to the Reff when a (new) VoC argument is passed. Expect this to change somewhat as new VoC are introduced and more complexity is needed.

A fixed date is set for the end of the second wave in NSW.

Some general house cleaning is done to remove data files from the git and add arguments to the bash scripts. 20000 sims are now run as too many simulations are being rejected in VIC.

tobinsouth committed:
Add Variant of Concern option to model.

Adding the ability to increase Reff starting at current date. Currently only the UK strain increase is implemented. Reff is increased by a random 1+Beta(6,14) multiplier for each day in a simulation.

tobinsouth committed:
Adding VoC param to run scripts

Adding the ability to run a set of simulations with a Variant of Concern increasing the Reff.

tobinsouth committed:
Update the simulation count on VoC run

tobinsouth committed:
Increasing limit of max cases

Max limit on cases was dropping valdi sims and shrinking the CI of the estimates.

tobinsouth committed:
Adding Reff increase to plots.

Adding the increase from VoC to Reff in the plots produced.

tobinsouth committed:
Increase simulation count

tobinsouth committed:
Add a VoC CSV output

tobinsouth committed:
Ignore output CSVs in Git

tobinsouth committed:
Delete UoA_2020-10-06local_obs.csv

tobinsouth committed:
Fixing end date of second wave in NSW

tobinsouth committed:
Removing all old data

tdennisliu commented on analysis/cprs/generate_posterior.py:
```diff
@@ -303,11 +303,14 @@
-    else:
-        possible_end_date = pd.to_datetime("2020-06-01")
-    sec_end_date = possible_end_date.strftime('%Y-%m-%d')
+    sec_end_date = '2021-01-18'
```
Just be mindful that this is a global end date on the second wave, and at line 348 there are state specific date ranges. Don't know what is a better solution, but might lead to conflicts later.

tdennisliu commented on analysis/forecast_plots.py:
```diff
@@ -142,6 +144,14 @@ def read_in_Reff(forecast_R=None,R_I=None,file_date = "2020-04-01"):
+                # We do so but editing a slice of the data frame. Forgive me for my sins.
+                VoC_dates_to_apply_idx = df_forecast.index[pd.to_datetime(df_forecast.date, format='%Y-%m-%d') > pd.to_datetime(file_date)]
+                df_slice_after_VoC = df_forecast.iloc[VoC_dates_to_apply_idx, 8:]
+                df_forecast.iloc[VoC_dates_to_apply_idx , 8:] = df_slice_after_VoC*(beta.rvs(6,14, size = df_slice_after_VoC.shape) + 1)
```
Why are the 8th column? Might want to comment in that for future Tobin (or Tobin-replacement :) )

tobinsouth replied:
Comments added. Future Tobin sends his thanks :)

tdennisliu reviewed (approved):
Changes look good. Would recommend changing the README (and possibly adding comments to bash scripts) to reflect the additional optional argument for VoC flag. Josh will almost certainly ask when you need to handover about how to properly add it.

tobinsouth committed:
Change second wave end date by one day from discussion.

tobinsouth committed:
Save inbox from intermediate run spam

tobinsouth committed:
Pass VoC param to record_to_csv.py

tobinsouth committed:
Update pipeline runcode

tobinsouth committed:
Removed fixed axis plots

tobinsouth committed:
Change VoC Beta param & add new imports period

Changed Beta dist multiplier to Beta(14,20).

Added a new time period for estimating imports over. Previous imports were higher than current levels. Lasting effect of imports was driving a high probability of 100 person outbreak in NSW without any local cases.
Note: This may be changed soon to a rolling or monthly import period approach.

tobinsouth committed:
Update README with VoC changes..

Added some minor spelling changes.

tobinsouth committed:
Add 30 days periods of imports.

tobinsouth committed:
Move VoC condition to fix bug.

tobinsouth committed:
Revert "Add 30 days periods of imports."

This reverts commit ecb15a7897eb6ba191270a343f5633a3645e341a.

tobinsouth committed:
Adding in some prints to help with future debugging.

tobinsouth committed:
Update start date to 2020-12-01

Forecasts are taking too long and are dropping too many sims

tobinsouth committed:
Update start date to 2020-12-01

Forgot this file in previous commit.

tobinsouth committed:
Remove VSCode files.

tobinsouth committed:
Update with new start date.

tdennisliu reviewed (approved):
All looks good, happy for you to merge. Might want to merge this into master soon before it gets too large and too many features get added at once.
